src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -136,63 +141,64 @@ bool convert_precision(ov::pass::PassBase& pass,
         }
     };
 
-    auto convert_node_output_precision = [&](const std::shared_ptr<ngraph::Node>& node) {
-        bool res = false;
-        // Handle case with Constants as they can have consumers from other nGraph Function object
-        const auto constant = ov::as_type_ptr<opset10::Constant>(node);
-        const auto it = const_to_internal_output.find(node.get());
-        if (constant && constant->get_output_element_type(0) == from && it != const_to_internal_output.end()) {
-            return fuse_type_to_constant(node, to, it->second);
-        }
+    std::function<bool(const std::shared_ptr<Model>&, bool)> convert_function_precision =
+        [&](const std::shared_ptr<Model>& f, bool is_subgraph) mutable {
+            bool is_changed = false;
+            bool is_output_precision_changed = false;
+
+            auto node_is_replaced = [](const std::shared_ptr<Node>& node) {
+                const auto outputs = node->outputs();
+                bool has_consumers = std::all_of(outputs.begin(), outputs.end(), [](const Output<Node>& output) {
+                    return output.get_target_inputs().size() == 0;
+                });
+                return has_consumers && !(is_type<op::v0::Result>(node) || is_type<op::Sink>(node));
+            };
+
+            auto convert_node_output_precision = [&](const std::shared_ptr<ngraph::Node>& node,
+                                                     const precisions_map& precisions) {
+                bool res = false;
+                // Handle case with Constants as they can have consumers from other nGraph Function object
+                const auto constant = ov::as_type_ptr<opset10::Constant>(node);
+                const auto it = const_to_internal_output.find(node.get());
+                if (constant && it != const_to_internal_output.end()) {
+                    return fuse_type_to_constant(node, precisions, it->second);
+                }
 
-        for (const auto& output : node->outputs()) {
-            if (output.get_element_type() == from) {
                 // Check that node type exists in map and we can fuse type into node
                 const auto t2f_it = type_to_fuse.find(node->get_type_info());
                 if (t2f_it != type_to_fuse.end()) {
-                    res |= t2f_it->second(node, to, output.get_index());
+                    res = t2f_it->second(node, precisions);
                 }
-            }
-        }
-        return res;
-    };
+                if ((is_changed || is_output_precision_changed || res) && !node_is_replaced(node)) {
```

Comment:
Are you sure that we need to invoke `node_is_replaced`?
As I can see, the `ops` is updated on line 204 
![image](https://user-images.githubusercontent.com/21181928/218432009-650f76b8-ab0f-4d7e-881d-a093f671f800.png)
I believe the topological sort (get_ordered_ops) won't return nodes without consumers

Reply:
the node handler (called at line 170) can also replace the node. If that's the case we don't have to revalidate the old one and sometimes we even can't (e.g. if the new node is type relaxed the old one can throw during validate_and_infer_types